src/index.ts

```diff
@@ -0,0 +1,61 @@
+interface Field {
+  left: number;
+  top: number;
+  width: number;
+  height: number;
+  chars: string;
+}
+
+export type Fieldset = Field[];
+
+function getNumberFromWords(words: string[]): string {
+  if(!words.length){
+    throw new Error('Words cannot be an empty array');
+  }
+  const number = words.pop();
+  if(number && number.match(/[A-Za-z\d]/) !== null){
```

Comment:
`number` could be `undefined` or a string (and also is the last word from the `words`), and we test here whether or not it contains a sensible value for an invoice number (this could be tweaked but should cover most invoice number formats).

If we have a match then we know we have the invoice number.